docs/getting_started.md

```diff
@@ -171,233 +208,453 @@ Also adding optimizer & model to this state has the advantage that we only need
+ax1.set_title('Loss')
+ax2.set_title('Accuracy')
+for dataset in ('train','test'):
+  ax1.plot(metrics_history[f'{dataset}_loss'], label=f'{dataset}_loss')
+  ax2.plot(metrics_history[f'{dataset}_accuracy'], label=f'{dataset}_accuracy')
+ax1.legend()
+ax2.legend()
+plt.show()
+plt.clf()
 ```
 
-+++ {"id": "UqNrWu7kIC9S"}
++++ {"id": "qQbKS0tV3sZ1"}
 
-## 14. Train and evaluate
+## 14. Inference on test set
 
-Once the training and testing is done after 10 epochs, the output should show that your model was able to achieve approximately 99% accuracy.
+Use the learned parameters to do model inference on the test set and visualize the images and their corresponding predicted labels.
 
 ```{code-cell}
-:id: 0nxgS5Z5IsT_
+---
+executionInfo:
+  elapsed: 98
+  status: ok
+  timestamp: 1671500871612
+  user:
+    displayName: Marcus Chiam
+    userId: '17531616275590396120'
+  user_tz: 300
+id: DFwxgBQf44ks
+---
+for test_batch in test_ds.as_numpy_iterator():
+  test_state = compute_metrics(state=test_state, batch=test_batch)
+  pred = state.apply_fn({'params': state.params}, test_batch['image']) # model inference
+  break # get only the first batch
+pred = pred.argmax(axis=1) # argmax the logits to get predicted labels
+```
 
-num_epochs = 10
-batch_size = 32
+```{code-cell}
+---
+executionInfo:
+  elapsed: 58
+  status: ok
+  timestamp: 1671500871794
+  user:
+    displayName: Marcus Chiam
+    userId: '17531616275590396120'
+  user_tz: 300
+id: 6482aa3G4E0T
+---
+def show_img(img, ax=None, title=None):
+  """Shows a single image."""
+  if ax is None:
+    ax = plt.gca()
+  ax.imshow(img[..., 0], cmap='gray')
+  ax.set_xticks([])
+  ax.set_yticks([])
+  if title:
+    ax.set_title(title)
+
+def show_img_grid(imgs, titles):
+  """Shows a grid of images."""
+  n = int(np.ceil(len(imgs)**.5))
+  _, axs = plt.subplots(n, n, figsize=(3 * n, 3 * n))
+  for i, (img, title) in enumerate(zip(imgs, titles)):
+    show_img(img, axs[i // n][i % n], title)
 ```
 
 ```{code-cell}
 ---
 colab:
-  base_uri: https://localhost:8080/
-id: ugGlV3u6Iq1A
-outputId: d0944ddb-8d5d-4e9f-9727-040789ef3f17
----
-for epoch in range(1, num_epochs + 1):
-  # Use a separate PRNG key to permute image data during shuffling
-  rng, input_rng = jax.random.split(rng)
-  # Run an optimization step over a training batch
-  state = train_epoch(state, train_ds, batch_size, epoch, input_rng)
-  # Evaluate on the test set after each training epoch
-  test_loss, test_accuracy = eval_model(state.params, test_ds)
-  print(' test epoch: %d, loss: %.2f, accuracy: %.2f' % (
-      epoch, test_loss, test_accuracy * 100))
+  height: 866
+executionInfo:
+  elapsed: 981
+  status: ok
+  timestamp: 1671500872908
+  user:
+    displayName: Marcus Chiam
+    userId: '17531616275590396120'
+  user_tz: 300
+id: 5d5nF3u44JFI
+outputId: 22e013f6-b9b7-4088-84f3-caaf248377ce
+---
+show_img_grid(
+    [test_batch['image'][idx] for idx in range(25)],
+    [f'label={pred[idx]}' for idx in range(25)],
+)
```

Comment:
All of this can be reduced to:

```python
fig, axs = plt.subplots(5, 5, figsize=(12, 12))
for i, ax in enumerate(axs.flatten()):
    ax.imshow(test_batch['image'][i, ..., 0], cmap='gray')
    ax.set_title(f"label={pred['label'][i]}")
    ax.axis('off')
```